src/hooks/useOnboardingForm.tsx

```diff
@@ -0,0 +1,41 @@
+"use client";
+import { useState } from "react";
+
+export enum FormSteps {
+  REGISTRATION,
+  ADDRESS,
+  CONTACT,
+  ABOUT_YOU,
+  PROFESSIONAL_EXP,
+}
+
+export function useOnboardingForm(initialStep = FormSteps.REGISTRATION) {
+  const [step, setStep] = useState<FormSteps>(initialStep);
+  const state = useOnboardingFormStates();
+  const canMovePrevious = step !== FormSteps.REGISTRATION;
+  const canMoveNext = step !== FormSteps.PROFESSIONAL_EXP && state.valid;
+
+  function moveNextStep() {
+    if (!canMoveNext) return;
+    setStep((step) => step + 1);
+  }
+
+  function movePrevStep() {
+    if (!canMovePrevious) return;
+    setStep((step) => step - 1);
+  }
+
+  return {
+    step,
+    state,
+    canMoveNext,
+    canMovePrevious,
+    moveNextStep,
+    movePrevStep,
+  };
+}
+
+function useOnboardingFormStates() {
+  // TODO: zod + form states
```

Comment:
[this](https://www.react-hook-form.com/api/useformcontext/) could help

Reply:
Sweet! Great tip @vzsoares 🙏🏼

Reply:
Hey Zenha, great suggestion. Thanks a lot for the review. We gonna solve this in a next PR just for form validation!